### PR TITLE
Move octokit to NuKeeper.GitHub

### PR DESCRIPTION
**Pull request by shep1987**
This PR removes the Octokit dependancy from the "NuKeeper" Project and moves it into Nukeeper.Github.

First commit is the bulk of the change, second is renaming everything. There are still bits of GitHub Functionality to move into the new project but i feel this is a good first step.

### CODE REVIEW

**Comment by AnthonySteele** [comment]
A good start, I think.
Will still have to look at places where `NuKeeper.GitHub` depends on `NuKeeper.Inspection`

File: NuKeeper.Tests/Engine/GitHubEngineTests.cs [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+using NuKeeper.Abstractions.DTOs;
 using NuKeeper.Abstractions.Logging;
 using NuKeeper.Engine;
-using NuKeeper.GitHub;
 using NuKeeper.Inspection.Files;
 using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NuKeeper.Abstractions.CollaborationPlatform;
```
Comment (skolima) [inline NuKeeper.Tests/Engine/GitHubEngineTests.cs]: Is using sorting disabled?

Comment (shep1987) [reply NuKeeper.Tests/Engine/GitHubEngineTests.cs]: looks like when using resharper to refactor it doesn't re-order afterwards... will go through and fix that now

**Comment by AnthonySteele** [comment]
Is this ready to merge now?

**Pull request merged**